spec/solver_spec.rb

```diff
@@ -0,0 +1,19 @@
+require './solver'
+
+RSpec.describe 'Solver' do
+  it 'generates the factorial of a number' do
+    solver = Solver.new
+    factor = solver.factoria(5)
+    expect(factor).to eq 120
+  end
```

Comment:
## The following are the test cases which are missing to cover, kindly have them covered so that we can move on with the project:
- [ ] Factorial for a Negative value should raises an exception as an expectation
- [ ] A Special case - when input is 0, the result should equal 1

Reply:
This issue is well resolved! ✔️